### PR TITLE
Make pill @ and matrix.to links behave the same

### PR DESCRIPTION
**Pull request by toger5**
Another go at: https://github.com/matrix-org/matrix-react-sdk/pull/7550
The idea is to convert all links to matrix.to links.
This sadly breaks pills since they are expecting a local url. So we change all the pill related code to use and expect a matrix to permalink as well.

I wont bother about the tests until we decide, that this is actually the right approach

---
Here's what your changelog entry will look like:

## ✨ Features
 * Make pill @ and matrix.to links behave the same ([\#7559](https://github.com/matrix-org/matrix-react-sdk/pull/7559)). Contributed by @toger5.

Preview: https://61e7114c58384814d8cb5ac8--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/utils/permalinks/Permalinks.ts [hunk 1]
```diff
@@ -331,8 +331,8 @@ export function tryTransformEntityToPermalink(entity: string): string {
     if (entity[0] === '@') return makeUserPermalink(entity);
     if (entity[0] === '+') return makeGroupPermalink(entity);
 
-    // Then try and merge it into a permalink
-    return tryTransformPermalinkToLocalHref(entity);
+    // does not need conversion
+    return entity;
```
Comment (turt2live) [inline src/utils/permalinks/Permalinks.ts:335]: The documentation and change don't appear to match - it should return null if the entity is unparsable, but here we now assume that it's always valid.

Can we update the implementation and/or the docs please?

**Review by turt2live — COMMENTED** [review]
otherwise seems fine - just need to fix the tests.

**Review by turt2live — APPROVED** [review]
I think this is sane

**Comment by toger5** [comment]
This is not idea:
```
if (transformed !== href || // for matrix symbols e.g. @user:server.tdl
                    decodeURIComponent(href).match(ELEMENT_URL_PATTERN) || // for https:vector|riot...
                    decodeURIComponent(href).match(MATRIXTO_URL_PATTERN) // for matrix.to
                ) {
                ```
but it gets superseded by https://github.com/matrix-org/matrix-react-sdk/pull/7453 anyways.
If we can get https://github.com/matrix-org/matrix-react-sdk/pull/7453 mergeable quick enough it might be best to just skit this pr?

**Comment by toger5** [comment]
this review: https://github.com/matrix-org/matrix-react-sdk/pull/7559#pullrequestreview-855788064
was also applied to  https://github.com/matrix-org/matrix-react-sdk/pull/7453

**Comment by jryans** [comment]
> If we can get #7453 mergeable quick enough it might be best to just skit this pr?

Sure, seems fine to me.

**Pull request closed without merging**